XPlat/Shopify/iOS/source/Shopify.iOS/Extras.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Foundation;
+using ObjCRuntime;
+using PassKit;
+
+namespace Shopify.Buy
+{
+	//public static class BUYRuntime
+	//{
+	//	// extern NSSet * class_getBUYProperties (Class clazz) __attribute__((overloadable));
+	//	[DllImport ("__Internal")]
+	//	static extern NSSet class_getBUYProperties (Class clazz);
+	//}
+
+	partial class BuyClient
```

Comment:
I couldn't for these ones as the `[Async]` attribute only works well when it is a `void` and the last parameter is an `Action<TReturn, NSError>` or `Action<TReturn>`

Also, this covers multiple return types as well as cancellation support for `NSUrlSessionDataTask`.

I did use the `[Async]` attribute on the other members. This can of course be confirmed/opposed my our resident iOS binding experts... such as @SotoiGhost ;)